ably/realtime_channel.go

```diff
@@ -451,29 +320,27 @@ func (c *RealtimeChannel) notify(msg *proto.ProtocolMessage) {
 	case proto.ActionAttached:
 		c.state.syncSet(StateChanAttached, nil)
 		c.queue.Flush()
+		c.Presence.mu.Lock()
```

Comment:
I feel like it should be the responsibility of `RealtimePresence` to manage it's own mutex.